grand_challenge_forge/partials/example-algorithm/inference.py.j2

```diff
@@ -71,62 +83,66 @@ def run():
         print(f.read())
 
     # For now, let us make bogus predictions
-    {%- for ci in phase.algorithm_outputs %}
-    output_{{ ci.slug.replace("-", "_")}} =
-        {%- if ci | has_example_value %}
-            {%- if ci.example_value is string -%}
-                "{{ ci.example_value }}"
+    {%- for socket in interface["outputs"] %}
+    output_{{ socket.slug.replace("-", "_")}} =
+        {%- if socket | has_example_value %}
+            {%- if socket.example_value is string -%}
+                "{{ socket.example_value }}"
             {%- else -%}
-                {{ ci.example_value }}
+                {{ socket.example_value }}
             {%- endif %}
-        {%- elif ci | is_image %} numpy.eye(4, 2)
-        {%- elif ci | is_json %} {"content": "should match the required format"}
-        {%- elif ci | is_file %} "content: should match the required format"
+        {%- elif socket | is_image %} numpy.eye(4, 2)
+        {%- elif socket | is_json %} {"content": "should match the required format"}
+        {%- elif socket | is_file %} "content: should match the required format"
         {% endif %}
     {%- endfor %}
 
     # Save your output
-    {% for ci in phase.algorithm_outputs -%}
-    {% set py_slug = ci.slug.replace("-", "_") -%}
-    {% if ci | is_image -%}
+    {% for socket in interface["outputs"] -%}
+    {% set py_slug = socket.slug.replace("-", "_") -%}
+    {% if socket | is_image -%}
     write_array_as_image_file(
-        location=OUTPUT_PATH / "{{ ci.relative_path }}",
+        location=OUTPUT_PATH / "{{ socket.relative_path }}",
         array=output_{{ py_slug }},
     )
     {% endif -%}
-    {% if ci | is_json -%}
+    {% if socket | is_json -%}
     write_json_file(
-        location=OUTPUT_PATH / "{{ ci.relative_path }}",
+        location=OUTPUT_PATH / "{{ socket.relative_path }}",
         content=output_{{ py_slug }}
     )
     {% endif -%}
-    {% if ci | is_file -%}
+    {% if socket | is_file -%}
     write_file(
-        location=OUTPUT_PATH / "{{ ci.relative_path }}",
+        location=OUTPUT_PATH / "{{ socket.relative_path }}",
         content=output_{{ py_slug }}
     )
     {% endif -%}
     {% endfor %}
     return 0
-{%- if phase.algorithm_inputs | has_json %}
+{% endfor %}
 
+def get_interface_key():
+    inputs = load_json_file(
```

Comment:
Might be worth to add a comment here that explains that inputs.json is a system generated file that is passed to each job automatically and that it contains information about the passed inputs, and the required outputs.

Reply:
AFAIK it's just a serializer for CIVs over the inputs that gets dumped to a file.

Reply:
You are right, GC code snippet that creates inputs.json:

```python    
def _with_inputs_json(self, *, input_civs):
        """
        An iterator over all inputs along with the special inputs.json,
        which serialises the metadata for all the other inputs such as
        the socket description.
        """
        yield from input_civs

        serializer = ComponentInterfaceValueSerializer(input_civs, many=True)
        yield ComponentInterfaceValue(
            value=serializer.data,
            interface=ComponentInterface(
                relative_path="inputs.json", kind=InterfaceKindChoices.ANY
            ),
        )
```

Reply:
Could be an argument to also add an outputs.json. Feels like a formality at this point so we'll get to it when someone needs it, if ever.